src/marshaller/HipieDDL.js

```diff
@@ -782,8 +782,10 @@
     Visualization.prototype.setWidget = function (widget) {
         this.widget = widget;
         this.events.setWidget(widget);
-        var columns = this.source.getColumns();
-        this.widget.columns(columns);
+        if (this.widget.columns) {
```

Comment:
@GordonSmith are you testing that the function exists here? should we check typeof if that is the case?